src/components/ContentPreview/ContentPreview.js

```diff
@@ -270,8 +255,29 @@ class ContentPreview extends PureComponent<Props, State> {
      * @return {void}
      */
     componentDidUpdate(prevProps: Props, prevState: State): void {
-        if (this.shouldLoadPreview(prevProps, prevState)) {
+        const { fileId, token } = this.props;
+        const hasFileIdChanged = prevProps.fileId !== fileId;
+        const hasTokenChanged = prevProps.token !== token;
+
+        if (hasFileIdChanged) {
+            this.destroyPreview();
+            this.fetchFile(fileId);
+        } else if (this.shouldLoadPreview(prevState)) {
             this.loadPreview();
+        } else if (hasTokenChanged) {
+            this.updatePreviewToken();
+        }
+    }
+
+    /**
+     * Updates the access token used by preview library
+     *
+     * @param {boolean} shouldReload - true if preview should be reloaded
+     */
+    updatePreviewToken(shouldReload: boolean = false) {
+        if (this.preview) {
+            const { token } = this.props;
```

Comment:
BTW, normally when only 1 prop is being used, and not repeated, its fine to use `this.props.token`